content/docs/atom-guide/look-dev/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,669 @@
+      m_simulationTime = simulationTime;
+      ```
+   3. In `Scene::PrepareSceneSrg()`, set the value of `SceneSrg`'s `m_prevTime` using the index:
+   
+      ```cpp
+      if (m_timeInputIndex.IsValid())
+      {
+         m_srg->SetConstant(m_timeInputIndex, m_simulationTime);
+      }
+
+      if (m_prevTimeInputIndex.IsValid()) {
+         m_srg->SetConstant(m_prevTimeInputIndex, m_prevSimulationTime);
+      }
+      ```
+
+Now that we have the previous time set up, we can actually use it in the vertex shader for our motion vector shader.
+
+1. Open `MeshMotionVectorVegetationBending.azsli`.
+2. We want to perform bending on the vertex at the `IN.m_position` but at the previous frame time, with the previous world position. Edit the code inside the `o_color_isBound` conditional to do the exact same as we did before, but with the appropriate time the world position:
+   ```
+   if (o_color_isBound) {
+      float currentTime = SceneSrg::m_time;
+      float prevTime = SceneSrg::m_prevTime;
+
+      // Overall wind
+      float4 currentBending = setUpWindBending(currentTime, worldPosition);
+      float4 prevBending = setUpWindBending(prevTime, prevWorldPosition);
+
+      // Detail bending
+      float3 currentOutPosition = detailBending(IN.m_position, IN.m_normal, IN.m_optional_color, currentTime, worldPosition, currentBending.w);
+      float3 prevOutPosition = detailBending(IN.m_position, IN.m_normal, IN.m_optional_color, prevTime, prevWorldPosition, prevBending.w);
+
+      // Main bending
+      OUT.m_position.xyz = mainBending(currentOutPosition, currentBending);
+      float3 prevPosition = mainBending(prevOutPosition, prevBending);
+
+      worldPosition = mul(objectToWorld, float4(OUT.m_position.xyz, 1.0));
+      prevWorldPosition = mul(prevObjectToWorld, float4(prevPosition, 1.0));
+   }
+   ```
+3. Take a look at the pixel shader to see how the motion vector is calculated! There is no need to edit the pixel shader.
+
+Amazing, we have added everything we need to add for motion vectors! However, if you open the **Editor** and just view the tree, you'll see that there is no difference. We can, however, observe that the motion vector pass works by using [**ImGui**](../../user-guide/interactivity/physics/debugging/#debugging-with-the-imgui-tool).
+1. Open the **Editor** and press **CTRL-G** to enter gameplay mode.
+2. Press the **Home** key on your keyboard. This brings up the toolbar at the top.
+3. Select **Atom Tools** > **Pass Viewer**.
+4. In the pop-up **PassTree View**, enable **Preview Attachment** and **Show Pass Attachments**.
+5. In the **PassTree**, find *MotionVectorPass* > *MeshMotionVectorPass* and select the line with `CameraMotion`. 
+6. Look at your tree, and you should see the motion vectors, with the colors denoting the direction!
+
+{{< video src="/images/atom-guide/vegetation-bending-tutorial/motionvectortree.mp4" autoplay="true" loop="true" muted="true" width="100%" info="Video of tree motion vector." >}}
```

Comment:
Do you know why the motion vector image is flickering? It seems this might indicate some kind of instability in the frame time SRG data or something.

Reply:
After looking into it, I think the reason why the image is flickering between black/another color is because the amount of movement is very small, causing the color to (sometimes) show up black, since it thinks that that vertex isn't moving. When I tried scaling the motion vector to be very large, there's no flickering. 

With detail bending, the flickering shown is different (flickers between colors) because it moves quickly in changing directions. I think I'll replace this current video that doesn't have detail bending to the video with detail bending.

However, in general I needed to scale the motion vector to get any color to show up in the first place, which I'm not sure is good practice if it is to be used with TAA, but I'll look more into this also.

Reply:
Also you might try altering the precision of your functions to use double for the calculations and return type to see if it's a floating point precision issue.